src/SignalR/docs/specs/TransportProtocols.md

```diff
@@ -197,3 +205,78 @@ When data is available, the server responds with a body in one of the two format
 If the `id` parameter is missing, a `400 Bad Request` response is returned. If there is no connection with the ID specified in `id`, a `404 Not Found` response is returned.
 
 When the client has finished with the connection, it can issue a `DELETE` request to `[endpoint-base]` (with the `id` in the query string) to gracefully terminate the connection. The server will complete the latest poll with `204` to indicate that it has shut down.
+
+## Ack Protocol
+
+The ack protocol primarily consists of writing and reading framing around the data being sent and received.
+All sends need to start with a 24 byte frame. The frame is 2 12 byte base64 encoded values. The first base64 value when decoded is the length of the payload being sent (minus the framing) as an int64 value. The second base64 value when decoded is the ack ID as an int64 of how many bytes have been received from the other side so far.
```

Comment:
"The frame is 2 12 byte base64 encoded values" - that's unclear whether it is 12 bytes before vs after base-64; IMO something like "The frame consists of 2 64-bit values, all base-64 encoded for a total of 12 bytes". Possibly also warrants a comment on "sign" - I presume these values should never be negative, but do we expect consumes to value the MSB re ulong values? maybe just say that the MSB of each of the 64-bit values is reserved for future use?

needs to explicitly specify endianness; I suggest little-endian, but: whatever as long as explicit

Honestly I don't think we need 64-bit frames (the first number); IMO we should restrict that to 32-bit (31-bit if we reserve the MSB).